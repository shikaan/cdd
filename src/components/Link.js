import React from 'react'
import {Link as GatsbyLink, navigate as gastbyNavigate} from 'gatsby'
import styled from 'styled-components'

const UnstyledButton = styled.button`
  background:none !important;
  border:none; 
  padding:0!important;
  cursor: pointer;
  text-align: unset;
  
  &:focus {outline:0;}
`

const UnstyledLink = ({href, onClick, ...rest}) => {
  if (href) {
    return React.createElement('a', {href, ...rest})
  } else if (onClick) {
    return React.createElement(UnstyledButton, {onClick, ...rest})
  }

  return React.createElement(GatsbyLink, rest)
}

export const navigate = gastbyNavigate

const Link = styled(UnstyledLink)(({theme}) => `
  font-family: ${theme.typography.primaryFont.fontFamily};
  font-weight: 700;
  text-decoration: none;
`)

export default Link
