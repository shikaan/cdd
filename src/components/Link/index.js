import React from 'react'
import { Link as GatsbyLink } from 'gatsby'
import styled from 'styled-components'

const UnstyledButton = styled.button`
  background:none !important;
  border:none; 
  padding:0!important;
  cursor: pointer;
  
  &:focus {outline:0;}
`

const Link = ({ href, onClick, children, ...rest }) => {
  if (href) {
    return React.createElement('a', { children, href, ...rest })
  } else if (onClick) {
    return React.createElement(UnstyledButton, { children, onClick, ...rest })
  }

  return React.createElement(GatsbyLink, { children, ...rest })
}

export default styled(Link)`
  text-decoration: none;
`
