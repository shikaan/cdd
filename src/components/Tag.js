import React from 'react';
import styled from 'styled-components'
import PropTypes from 'prop-types'
import {camelCase, upperFirst} from 'lodash'

import Link from '~components/Link'

const StyledLink = styled(Link)(({theme}) => `
  font-family: ${theme.typography.primaryFont.fontFamily};
  font-size: ${theme.typography.baseFontSize.multiply(0.75)};
  font-weight: 700;
  text-decoration-line: underline;
  padding: 0 ${theme.typography.baseFontSize.multiply(0.5)} 0 0;
  white-space: nowrap;
  display: inline-block;
  color: ${theme.color.black};
`)

const Tag = React.memo(({children, ...rest}) => {
  return <StyledLink children={`#${upperFirst(camelCase(children))}`} {...rest} />
})

Tag.propTypes = {
  children: PropTypes.string
}

export default Tag;
