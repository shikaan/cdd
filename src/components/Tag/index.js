import styled from 'styled-components'

import Link from '~components/Link'

export default styled(Link)`
  font-family: ${({ theme }) => theme.typography.primaryFont.fontFamily};
  font-size: ${({ theme }) => theme.typography.baseFontSize.multiply(0.75)};
  font-weight: 700;
  text-decoration-line: underline;
  padding: 0 ${({ theme }) => theme.typography.baseFontSize.multiply(0.5)} 0 0;
  text-transform: capitalize;
  white-space: nowrap;
  display: inline-block;

  &:before {
    content: '#'
  }
`
