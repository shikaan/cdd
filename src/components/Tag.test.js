import React from "react"
import 'jest-styled-components'
import {shallow} from "enzyme";

import Tag from "./Tag"
import themeMock from '~theme'
import Link from "./Link"

describe("Tag", () => {
  it("renders correctly", () => {
    const wrapper = shallow(
      <Tag theme={themeMock}>text</Tag>
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('renders an anchor', () => {
    const wrapper = shallow(
      <Tag theme={themeMock}>text</Tag>
    )

    expect(wrapper.type().displayName).toContain(Link.displayName)
  })

  describe('renders text in PascalCase prefixed by #', () => {
    it('one word', () => {
      const wrapper = shallow(
        <Tag theme={themeMock}>text</Tag>
      )

      expect(wrapper.text()).toEqual('#Text')
    })

    it('more words with spaces', () => {
      const wrapper = shallow(
        <Tag theme={themeMock}>text one</Tag>
      )

      expect(wrapper.text()).toEqual('#TextOne')
    })

    it('more words with dashes', () => {
      const wrapper = shallow(
        <Tag theme={themeMock}>text-one</Tag>
      )

      expect(wrapper.text()).toEqual('#TextOne')
    })

    it('more words with underscores', () => {
      const wrapper = shallow(
        <Tag theme={themeMock}>text_one</Tag>
      )

      expect(wrapper.text()).toEqual('#TextOne')
    })
  })
})
