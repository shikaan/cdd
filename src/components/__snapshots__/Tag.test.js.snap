// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag renders correctly 1`] = `
<Styled(Styled(UnstyledLink))
  theme={
    Object {
      "breakpoint": Object {
        "lg": "1184px",
        "md": "980px",
        "mg": "1920px",
        "sm": "768px",
        "xl": "1366px",
        "xs": "480px",
      },
      "color": Object {
        "black": "rgba(0, 0, 0, .82)",
        "coffee": "rgba(145, 107, 85, 1)",
        "darkGrey": "rgba(79, 79, 79, 1)",
        "grey": "rgba(0, 0, 0, .32)",
        "lightCoffee": "rgba(249, 245, 242, 1)",
        "lightGrey": "rgba(0, 0, 0, .05)",
        "mediumGrey": "rgba(0, 0, 0, .38)",
        "paleGrey": "rgba(242, 242, 242, 1)",
        "white": "rgba(255, 255, 255, 1)",
      },
      "typography": Object {
        "__esModule": true,
        "baseFontSize": Size {
          "_rank": 2,
        },
        "baseLineHeight": "normal",
        "primaryFont": Object {
          "fontFamily": "Merriweather, Georgia, serif",
          "hover": "rgba(0, 0, 0, .82)",
        },
        "secondaryFont": Object {
          "fontFamily": "Lato, Geneva, Tahoma, sans-serif",
          "hover": "rgba(0, 0, 0, .82)",
        },
      },
      "zIndexRank": Object {
        "0": "0",
        "1": "100",
        "2": "200",
        "3": "300",
        "4": "400",
        "5": "500",
        "6": "600",
        "7": "700",
        "8": "800",
        "9": "900",
        "bottom": "-100",
        "top": "1000",
      },
    }
  }
>
  #Text
</Styled(Styled(UnstyledLink))>
`;
